validator/schema.go

```diff
@@ -293,6 +293,15 @@ func validateDirectives(schema *Schema, dirs DirectiveList, currentDirective *Di
 		if schema.Directives[dir.Name] == nil {
 			return gqlerror.ErrorPosf(dir.Position, "Undefined directive %s.", dir.Name)
 		}
+		argMap := make(map[string]int)
+		for i, arg := range schema.Directives[dir.Name].Arguments {
+			argMap[arg.Name] = i
+		}
+		for _, arg := range dir.Arguments {
+			if _, ok := argMap[arg.Name]; !ok {
+				return gqlerror.ErrorPosf(dir.Position, "Argument %s is not valid for directive %s", arg.Name, dir.Name)
+			}
```

Comment:
Also check that the type of the argument is the same as what was declared in the schema. So, if `@foo(bla: Int!)` was declared and someone was writing `@foo(bla: "I can send String for Int")`, then that should also give an error saying 
```
Argument %s for directive %s expects value of kind `Int`, got: `String`.
```
And add test cases in [validator/schema_test.yml](https://github.com/dgraph-io/gqlparser/blob/master/validator/schema_test.yml).